### PR TITLE
Add unit test for secret backend invocation from AD

### PR DESCRIPTION
**Pull request by L3n41c**
### What does this PR do?

Add unit test for secret backend invocation from AD.

### Motivation

Improve test coverage.

### Describe your test plan

Launch the unit tests.

### CODE REVIEW

**Review by ahmed-mez — APPROVED** [review]

File: pkg/autodiscovery/autoconfig_test.go [hunk 1]
```diff
@@ -384,3 +386,97 @@ func TestCheckOverride(t *testing.T) {
...
+			{
+				expectedData:   []byte("param2: ENC[bar]"),
+				expectedOrigin: "cpu",
+				returnedData:   []byte("param2: bar"),
+				returnedError:  nil,
+			},
+		},
+	}
+
+	secretsDecrypt = mockDecrypt.getDecryptFunc()
```
Comment (hush-hush) [inline pkg/autodiscovery/autoconfig_test.go:465]: You need to reset it back to normal after the test.

**Review by hush-hush — CHANGES_REQUESTED** [review]

File: pkg/autodiscovery/autoconfig_test.go (hunk 1 above)
Comment (L3n41c) [reply pkg/autodiscovery/autoconfig_test.go:466]: Good catch ! Thanks !
Here is it: https://github.com/DataDog/datadog-agent/pull/6914/commits/af8931d5832b05d9e55715800e41133a9f28c91b

File: pkg/autodiscovery/autoconfig.go [hunk 2]
```diff
@@ -33,6 +33,8 @@ var (
 	errorStats            = newAcErrorStats()
 )
 
+var secretsDecrypt func(data []byte, origin string) ([]byte, error)
```
Comment (hush-hush) [inline pkg/autodiscovery/autoconfig.go:36]: Why not directly assigned it ?

```go
var secretsDecrypt = secrets.Decryt
```

**Review by hush-hush — COMMENTED** [review]
On last comment I missed last time. Looks good to go after that.

File: pkg/autodiscovery/autoconfig.go (hunk 2 above)
Comment (L3n41c) [reply pkg/autodiscovery/autoconfig.go]: You’re definitively right ! => https://github.com/DataDog/datadog-agent/pull/6914/commits/7cfd779f181452c9615c3b192b7d5996abdaef85

**Review by hush-hush — APPROVED** [review]

**Pull request merged**